README.md
## Description

Installs/Configures [Atlassian Bamboo](https://www.atlassian.com/software/bamboo/) server.

## Requirements

### Platforms

* CentOS 6
* RHEL 6
* Ubuntu 12.04, 12.10, 13.04

### Chef

* The cookbook requires Chef 11 due to `ark` usage

### Databases

* HSQLDB (not recommended for production usage)
* MySQL
* Postgres
* Percona

### Cookbooks

Required [Opscode Cookbooks](https://github.com/opscode-cookbooks/)

* [apache2](https://github.com/opscode-cookbooks/apache2) (if using Apache 2 proxy)
* [ark](https://github.com/opscode-cookbooks/ark)
* [database](https://github.com/opscode-cookbooks/database)
* [java](https://github.com/opscode-cookbooks/java)
* [mysql](https://github.com/opscode-cookbooks/mysql) (if using MySQL database)
* [postgresql](https://github.com/opscode-cookbooks/postgresql) (if using Postgres database)

Third-Party Cookbooks

* [mysql_connector](https://github.com/bflad/chef-mysql_connector) (if using MySQL/Percona database)
* [percona](https://github.com/phlipper/chef-percona) (if using Percona database)

## Attributes

These attributes are under the `node['bamboo']` namespace.

Attribute | Description | Type | Default
----------|-------------|------|--------
checksum | SHA256 checksum for Bamboo install | String | auto-detected (see attributes/default.rb)
home_path | home data directory for Bamboo user | String | /var/atlassian/application-data/bamboo
install_path | location to install Bamboo | String | /opt/atlassian
install_type | Bamboo install type - "standalone" only for now | String | standalone
url_base | URL base for Bamboo install | String | http://www.atlassian.com/software/bamboo/downloads/binary/atlassian-bamboo
url | URL for Bamboo install | String | auto-detected (see attributes/default.rb)
user | user to run Bamboo | String | bamboo
version | Bamboo version to install | String | 5.5.1

### Bamboo Database Attributes

All of these `node['bamboo']['database']` attributes are overridden by `bamboo/bamboo` encrypted data bag (Hosted Chef) or data bag (Chef Solo), if it exists

Attribute | Description | Type | Default
----------|-------------|------|--------
host | FQDN or "localhost" (localhost automatically installs `['database']['type']` server) | String | localhost
name | Bamboo database name | String | bamboo
password | Bamboo database user password | String | changeit
port | Bamboo database port | Fixnum | 3306
testInterval | Bamboo database pool idle test interval in minutes | Fixnum | 2
type | Bamboo database type - "hsqldb" (not recommended), "mysql", "percona" or "postgresql" | String | mysql
user | Bamboo database user | String | bamboo

### Bamboo JVM Attributes

These attributes are under the `node['bamboo']['jvm']` namespace.

Attribute | Description | Type | Default
----------|-------------|------|--------
minimum_memory | JVM minimum memory | String | 512m
maximum_memory | JVM maximum memory | String | 768m
maximum_permgen | JVM maximum PermGen memory | String | 256m
java_opts | additional JAVA_OPTS to be passed to Bamboo JVM during startup | String | ""
support_args | additional JAVA_OPTS recommended by Atlassian support for Bamboo JVM during startup | String | ""

### Bamboo Tomcat Attributes

These attributes are under the `node['bamboo']['tomcat']` namespace.

Any `node['bamboo']['tomcat']['key*']` attributes are overridden by `bamboo/bamboo` encrypted data bag (Hosted Chef) or data bag (Chef Solo), if it exists

Attribute | Description | Type | Default
----------|-------------|------|--------
keyAlias | Tomcat SSL keystore alias | String | tomcat
keystoreFile | Tomcat SSL keystore file - will automatically generate self-signed keystore file if left as default | String | `#{node['bamboo']['home_path']}/.keystore`
keystorePass | Tomcat SSL keystore passphrase | String | changeit
port | Tomcat HTTP port | Fixnum | 8085
ssl_port | Tomcat HTTPS port | Fixnum | 8443
context | Tomcat context | String | ""

## Recipes

* `recipe[bamboo]` Installs Atlassian Bamboo with built-in Tomcat and Apache 2 proxy
* `recipe[bamboo::apache2]` Installs/configures Apache 2 proxy for Bamboo (ports 80/443)
* `recipe[bamboo::configuration]` Configures Bamboo's settings
* `recipe[bamboo::database]` Installs/configures MySQL/Percona/Postgres server, database, and user for Bamboo
* `recipe[bamboo::linux_standalone]` Installs/configures Bamboo via Linux standalone archive
* `recipe[bamboo::service_init]` Installs/configures Bamboo init service
* `recipe[bamboo::tomcat_configuration]` Configures Bamboo's built-in Tomcat

## Contributing

Please use standard Github issues/pull requests.

## License and Contributors

Please see license information in: [LICENSE](LICENSE)

* Sergio Leone <sergio.leone@qvc.com>


CHANGELOG.md
#0.5.0

* Enhancement: Default to Bamboo 5.8.1

#0.4.2

* Enhancement: Default to Bamboo 5.7.2

#0.4.1

* BugFix: Fixed version from 5.7 to 5.7.0

# 0.4.0

* Enhancement: Default to Bamboo 5.7

# 0.3.0

* Enhancement: Default to Bamboo 5.6.1

# 0.2.0

* Enhancement: Default to Bamboo 5.6.0

# 0.1.0

Initial release of chef-bamboo


attributes/default.rb
default['bamboo']['home_path']    = '/var/atlassian/application-data/bamboo'
default['bamboo']['install_path'] = '/opt/atlassian'
default['bamboo']['install_type'] = 'standalone'
default['bamboo']['service_type'] = 'init'
default['bamboo']['url_base']     = 'http://www.atlassian.com/software/bamboo/downloads/binary/atlassian-bamboo'
default['bamboo']['user']         = 'bamboo'
default['bamboo']['version']      = '5.8.1'

default['bamboo']['url']      = "#{node['bamboo']['url_base']}-#{node['bamboo']['version']}.tar.gz"
default['bamboo']['checksum'] =
case node['bamboo']['version']
	when '5.5.1' then '1a9229158d7347d8debeca824ce1511d40e1baa8514309aea576e0c07d73cf19'
	when '5.6.0' then '71c72e094fbcc258aa0f0cf74c652e0d63887c35b900f563bc672db99c5d9b98'
	when '5.6.1' then '28fcf425b44737fbe7aedc9cdccacee788d174c3cd6cc2beb4ce5f3ca5d0de27'
	when '5.6.2' then '57737a00207642e7be4ceea7b702e30e1caccaaa51e08699c2c9ae588e4c88b3'
	when '5.7.0' then 'd5de9b25502f78a2b975a74f9200a3ac15f91f1452e584344621cdd60368a690'
	when '5.7.2' then 'b378500ea61803333fc27b0a42cfbdfbce5dccc75cead0b464dbc4e5e0bddb17'
	when '5.8.1' then 'bb691d22273ceaa999eb435dfb4e8d697c3c803505218845bf5785ec2785cbd8'
end

default['bamboo']['apache2']['webapp_enable']      = true
default['bamboo']['apache2']['access_log']         = ''
default['bamboo']['apache2']['error_log']          = ''
default['bamboo']['apache2']['port']               = 80
default['bamboo']['apache2']['virtual_host_alias'] = node['fqdn']
default['bamboo']['apache2']['virtual_host_name']  = node['hostname']

default['bamboo']['apache2']['ssl']['access_log']       = ''
default['bamboo']['apache2']['ssl']['chain_file']       = ''
default['bamboo']['apache2']['ssl']['error_log']        = ''
default['bamboo']['apache2']['ssl']['port']             = 443

case node['platform_family']
when 'rhel'
  default['bamboo']['apache2']['ssl']['certificate_file'] = '/etc/pki/tls/certs/localhost.crt'
  default['bamboo']['apache2']['ssl']['key_file']         = '/etc/pki/tls/private/localhost.key'
else
  default['bamboo']['apache2']['ssl']['certificate_file'] = '/etc/ssl/certs/ssl-cert-snakeoil.pem'
  default['bamboo']['apache2']['ssl']['key_file']         = '/etc/ssl/private/ssl-cert-snakeoil.key'
end

default['bamboo']['database']['host']     = 'localhost'
default['bamboo']['database']['name']     = 'bamboo'
default['bamboo']['database']['password'] = 'changeit'
default['bamboo']['database']['port']     = 3306
default['bamboo']['database']['testInterval'] = 2
default['bamboo']['database']['type']     = 'mysql'
default['bamboo']['database']['user']     = 'bamboo'

default['bamboo']['jvm']['minimum_memory']  = '512m'
default['bamboo']['jvm']['maximum_memory']  = '768m'
default['bamboo']['jvm']['maximum_permgen'] = '256m'
default['bamboo']['jvm']['java_opts']       = ''
default['bamboo']['jvm']['support_args']    = ''

default['bamboo']['tomcat']['keyAlias']     = 'tomcat'
default['bamboo']['tomcat']['keystoreFile'] = "#{node['bamboo']['home_path']}/.keystore"
default['bamboo']['tomcat']['keystorePass'] = 'changeit'
default['bamboo']['tomcat']['port']         = '8085'
default['bamboo']['tomcat']['ssl_port']     = '8443'
default['bamboo']['context']                = ''

default['bamboo']['crowd']['integration'] = false
default['bamboo']['crowd']['sso'] = false
default['bamboo']['crowd']['application.name'] = 'bamboo'
default['bamboo']['crowd']['application.password'] = 'password'
default['bamboo']['crowd']['application.login.url'] = 'http://localhost:8095/crowd/console/'
default['bamboo']['crowd']['crowd.server.url'] = 'http://localhost:8095/crowd/services'
default['bamboo']['crowd']['crowd.base.url'] = 'http://localhost:8095/crowd/'
default['bamboo']['crowd']['session.validationinterval'] = 2
